### PR TITLE
chore(planning/control/vehicle): declare ROS params in yaml files

### PR DESCRIPTION
**Pull request by yuki-takagi-66**
## Description
In order to ready for JSON scheme, change ROS params to declare in yaml files.
launch: https://github.com/autowarefoundation/autoware_launch/pull/833

the work for raw_vehicle_cmd_converter is still remain.

## Tests performed

Not yet

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by rej55 — APPROVED** [review]

**Review by TakaHoribe — APPROVED** [review]

**Review by satoshi-ota — APPROVED** [review]

File: control/joy_controller/config/joy_controller.param.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /**:
   ros__parameters:
-    joy_type: DS4
+    joy_type: $(var joy_type)
```
Comment (xmfcx) [inline control/joy_controller/config/joy_controller.param.yaml:3]: How is this `$(var joy_type)` resolved?

This non-resolved variable causes the CI for this to fail:
https://github.com/autowarefoundation/autoware.universe/actions/runs/7981506848/job/21793322599#step:3:45

```
🔍 Validating: control/joy_controller/config/joy_controller.param.yaml 🆚 control/joy_controller/schema/joy_controller.schema.json ❌ Failed
Schema validation errors were encountered.
  control/joy_controller/config/joy_controller.param.yaml::$./**.ros__parameters.joy_type: '$(var joy_type)' is not one of ['P65', 'DS4', 'G29', 'XBOX']
```

Is there a specific reason to keep this as `$(var joy_type)` or could you revert this back to a default to comply with the schema file?

@yuki-takagi-66 @kaspermeck-arm

Comment (xmfcx) [reply control/joy_controller/config/joy_controller.param.yaml:3]: Slightly (but not much) related to:
- https://github.com/autowarefoundation/autoware-github-actions/pull/285

Comment (xmfcx) [reply control/joy_controller/config/joy_controller.param.yaml:3]: Fixing this in:
- https://github.com/autowarefoundation/autoware.universe/pull/6474

Comment (kaspermeck-arm) [reply control/joy_controller/config/joy_controller.param.yaml:3]: @xmfcx 

Just looking into this, find
* https://github.com/autowarefoundation/autoware-github-actions/issues/232#issuecomment-1857640095

regarding how to use environmental variables. Is `joy_type` really an environmental variable like `$HOME`?

@oguzkaganozt @ambroise-arm

Comment (kaspermeck-arm) [reply control/joy_controller/config/joy_controller.param.yaml:3]: @xmfcx 

Thanks for creating the PR! Just added a minor comment.

**Pull request merged**